### ISSUE TITLE
Publish to NPM

### ISSUE THREAD

**Issue by akinnee-gl** [mstefaniuk/graph-viz-d3-js#18]
It would be awesome if this were published to NPM!

**Comment by mstefaniuk** [comment]
It is published in Bower. Could you convince me to publish to NPM in addition?

**Comment by akinnee-gl** [comment]
I had to do a lot of extra work to get it working in Browserify or Webpack. Lots of developers use those tools.

**Comment by mstefaniuk** [comment]
It looks not so complicated. Now dependencies of the component are managed by Bower and they are:
```json
  "dependencies": {
    "requirejs": "2.1.15",
    "d3": "3.5.3",
    "requirejs-web-workers": "~1.0.1",
    "viz.js": "~1.3.0"
  }
```
Should I put these in `package.json`? As I checked there is no `require.js` worker plugin so it maybe forces some bigger changes.

**Comment by akinnee-gl** [comment]
Installing those via npm should work. I also had to use a worker loader to make it work in webpack instead of the requirejs loader you have in your repo. https://www.npmjs.com/package/worker-loader

**Comment by mstefaniuk** [comment]
Could you create a pull request for that?

**Comment by akinnee-gl** [comment]
Worker-loader is something the developer would add to their build process. Would you like me to add that to the documentation?

**Comment by mstefaniuk** [comment]
My component [uses worker plugin](/mstefaniuk/graph-viz-d3-js/blob/master/src/js/renderer.js#L1) to load parsing and rendering WebWorker. Do you suggest to replace one plugin with another by component end developer?

**Comment by akinnee-gl** [comment]
Yes. You are using the requirejs web worker plugin. Webpack and Browserify have a similar module called worker-loader.

Here is a list of the changes I had to make to get this module working with Webpack:

* change all require statements to relative paths
* install dependencies via npm: d3, viz.js, worker-loader

After these changes, no dependency paths need to be added to a config file.

**Comment by mstefaniuk** [comment]
Do it, please. Current build and test process relies on Bower so take it also into account.

**Comment by quiasmo** [comment]
Hey, an NPM package to use with webpack sounds good to me too. @akinnee-gl Is there anything you need besides encouragmenent to get that pull request? Perhaps you could share your branch?

**Comment by akinnee-gl** [comment]
Actually, because this doesn't support all of the functionality of Viz.js, I ended up just using Viz.js directly.

**Comment by tedwong** [comment]
@akinnee-gl  Would you kindly share your `webpack.config.js`?  I have a hand time on deploy it using webpack.

**Comment by akinnee-gl** [comment]
Personally I would recommend you just use Viz.js. It's less complicated and it's already on npm. If you really need to animate your svg, you can still do that using D3 separately.
